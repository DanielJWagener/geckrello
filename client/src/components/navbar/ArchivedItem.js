import React from "react";
import { connect } from "react-redux";
import { restoreList, restoreCard } from "../../actions";

const ArchivedItem = props => {
  const restore = () => {
    if (props.itemType === "list") {
      props.restoreList(props.itemId);
    } else if (props.itemType === "card") {
      props.restoreCard(props.itemId);
    }
  };

  return (
    <div className="archive__item">
      <div className="archive__item--title">{props.itemTitle}</div>
      <button className="archive__item--restore" onClick={restore}>
        Restore
      </button>
    </div>
  );
};

const mapStateToProps = state => {
  return { lists: state.lists, cards: state.cards };
};

export default connect(
  mapStateToProps,
  { restoreList, restoreCard }
)(ArchivedItem);
