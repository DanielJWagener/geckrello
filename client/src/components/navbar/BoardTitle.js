import React, { Component } from "react";

class BoardTitle extends Component {
  state = {
    inputText: "Board Title",
    titleText: "Board Title"
  };

  // Initially sets size of input and page title
  componentDidMount() {
    this.setInputSize(this.state.titleText.length);
    this.setPageTitle();
  }

  // Sets page title and input size every time state is updated
  componentDidUpdate() {
    this.setPageTitle();
    this.setInputSize();
  }

  // Updates state on user input
  inputHandler = event => {
    const inputText = event.target.value;

    this.setState({
      inputText: inputText
    });
  };

  // When user input hears Enter or Escape, calls checkEmptyInput
  keyDownHandler = event => {
    const newTitle = event.target.value;

    if (event.key === "Enter" || event.key === "Escape") {
      this.checkEmptyInput(newTitle);
    }
  };

  // Calls checkEmptyInput when input loses focus
  onBlurHandler = event => {
    const newTitle = event.target.value;

    event.target.className = "navbar__input";

    this.checkEmptyInput(newTitle);
  };

  // Changes input styling on focus, and selects all input text
  onFocusHandler = event => {
    event.target.className = "navbar__input navbar__input--infocus";

    event.target.select();
  };

  // Changes input styling on mouse over
  onMouseOverHandler = event => {
    document.activeElement === event.target
      ? (event.target.className = "navbar__input navbar__input--infocus")
      : (event.target.className = "navbar__input navbar__input--hover");
  };

  // Changes input styling on mouse out
  onMouseOutHandler = event => {
    document.activeElement === event.target
      ? (event.target.className = "navbar__input navbar__input--infocus")
      : (event.target.className = "navbar__input");
  };

  // If user input is empty, resets value to most recently saved value of titleText
  // Otherwise, value of titleText set to current input value
  checkEmptyInput = newTitle => {
    if (newTitle === "") {
      this.setState({
        inputText: this.state.titleText
      });
    } else {
      this.setState({
        titleText: newTitle
      });
    }
  };

  // Changes input width dynamically
  setInputSize = () => {
    const input = document.querySelector("input");

    input.style.width = `${this.state.inputText.length * 10 + 30}px`;
  };

  // Changes page title dynamically
  setPageTitle = () => {
    document.title = `${this.state.inputText} - Geckorello, a Trello Front-End Clone`;
  };

  render() {
    return (
      <div>
        <input
          className="navbar__input"
          value={this.state.inputText}
          onChange={this.inputHandler}
          onKeyDown={this.keyDownHandler}
          onBlur={this.onBlurHandler}
          onFocus={this.onFocusHandler}
          onMouseOver={this.onMouseOverHandler}
          onMouseOut={this.onMouseOutHandler}
        ></input>
      </div>
    );
  }
}

export default BoardTitle;
