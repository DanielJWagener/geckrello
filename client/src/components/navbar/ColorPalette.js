import React from "react";

const ColorPalette = props => {
  const updateBackgroundColor = () => {
    let root = document.documentElement;

    root.style.setProperty("--body-background", props.color);
    root.style.setProperty("--body-background-light", props.colorLight);
  };
  return (
    <div
      className="color-palette"
      style={{
        borderRadius: "10px",
        height: "7.5rem",
        width: "14rem",
        backgroundColor: props.color,
        margin: "1.5rem",
        cursor: "pointer"
      }}
      onClick={updateBackgroundColor}
    ></div>
  );
};

export default ColorPalette;
