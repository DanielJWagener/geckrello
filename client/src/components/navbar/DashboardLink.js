import React from "react";
import { Link } from "react-router-dom";

const Home = () => {
  return (
    <div>
      <Link to="/dashboard" className="auth-link">
        Home
      </Link>
    </div>
  );
};

export default Home;
