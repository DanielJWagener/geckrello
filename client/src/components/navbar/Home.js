import React from "react";

const Home = () => {
  return (
    <div>
      <a href="/" className="auth-link">
        Home
      </a>
    </div>
  );
};

export default Home;
