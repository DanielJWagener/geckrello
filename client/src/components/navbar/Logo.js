import React from "react";

const logo = () => {
  return (
    <div className="logo navbar__logo">
      <h1>Geckorello</h1>
    </div>
  );
};

export default logo;
