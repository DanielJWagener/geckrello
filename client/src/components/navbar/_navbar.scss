.navbar {
  display: grid;
  grid-template-areas:
    "header header header"
    "title . menu";
  position: relative;

  &__logo {
    width: 100vw;
    padding: 5px 0;
    grid-area: header;
    font-weight: bold;
    font-size: 1.5em;
    color: white;
    text-align: center;
  }

  &__input {
    margin-left: 10px;
    padding: 5px 0;
    grid-area: title;
    background-color: var(--body-background);
    border: none;
    border-radius: 0.5rem;
    font-size: 2em;
    font-weight: 700;
    color: white;
    text-align: center;
  }

  &__input--infocus {
    background-color: white;
    color: $default-font-color;
  }

  &__input--hover {
    background-color: var(--body-background-light);
  }

  &__menu-wrapper {
    grid-area: menu;
    margin-right: 3rem;
  }

  &__menu {
    color: white;
    cursor: pointer;
    border-radius: 5px;

    padding: 5px;
    border: 2px solid var(--body-background-light);

    &--hover {
      background-color: var(--body-background-light);
    }
  }

  &__menu h1 {
    font-weight: 100;
  }
}
