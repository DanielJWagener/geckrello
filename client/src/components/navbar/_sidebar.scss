#sidebar-root {
  position: fixed;
  top: 9rem;
  right: calc(-1 * #{$sidebar-width});
  height: calc(var(--vh, 1vh) * 100 - 9rem);
  width: $sidebar-width;
  z-index: 2;

  @media only screen and (max-width: 400px) {
    width: 100vw;
  }
}

.portal-inner {
  height: 100%;
}

.sidebar {
  position: absolute;

  &--hidden {
    left: 0rem;
  }

  &--shown {
    left: calc(-1 * #{$sidebar-width});
  }

  height: calc(var(--vh, 1vh) * 100 - 9rem);
  background-color: $component-background;
  border-left: 2px solid white;
  border-radius: 8px 0 0 8px;
  width: $sidebar-width;
  padding: 1.5rem;
  transition: left 0.3s;

  @media only screen and (max-width: 400px) {
    width: 100vw;
  }

  &__wrapper {
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: $sidebar-width;
  }

  .sidebar-menu {
    &__heading {
      font-size: 2rem;
      font-weight: 700;
      text-align: center;
      margin-bottom: 1.5rem;
    }

    &__item {
      font-size: 2rem;
      padding: 2rem 1rem 2rem 1rem;
      width: 100%;
      text-align: left;
      border: none;
      border-radius: 3px;
      background-color: $component-background;
      cursor: pointer;
      &:hover {
        background-color: $component-background-dark;
      }
    }

    &__close {
      position: absolute;
      top: 3px;
      right: 1rem;
      cursor: pointer;
      font-size: 3rem;

      &:hover {
        color: red;
      }
    }

    &__interface {
      position: absolute;
      top: 0;
      width: $sidebar-width;

      height: calc(var(--vh, 1vh) * 100 - 9rem);
      overflow-y: auto;
      background-color: white;
      border-left: 2px solid $component-background;
      border-radius: 8px 0 0 8px;
      padding: 1.5rem;
      transition: all 0.3s;

      @media only screen and (max-width: 400px) {
        width: 100vw;
      }

      &--hidden {
        right: calc(-1 * #{$sidebar-width});
      }

      &--visible {
        right: 0;
      }

      &--close {
        position: absolute;
        top: 3px;
        right: 1rem;
        cursor: pointer;
        font-size: 3rem;

        &:hover {
          color: red;
        }
      }

      .archive {
        &-toggle {
          display: flex;
          justify-content: center;

          &__button {
            width: 25%;
            font-size: 2rem;
            color: white;
            padding: 0.5rem 1rem 0.5rem 1rem;
            border-radius: 5px;
            border: none;
            margin: 0.5rem;

            &:focus {
              outline: none;
            }

            &--inactive {
              background-color: #444;
              cursor: pointer;
            }

            &--active {
              background-color: blue;
            }
          }
        }

        &__list {
          margin-top: 2rem;
        }

        &__item {
          display: flex;
          justify-content: space-around;
          align-items: flex-center;
          margin-bottom: 2rem;

          &--title {
            font-size: 2rem;
            padding: 1em 4rem 1rem 4rem;
            text-align: center;
            background-color: #ddd;
            border-radius: 5px;
            width: 50%;
          }

          &--restore {
            font-size: 2rem;
            padding: 0.5rem 1rem 0.5rem 1rem;
            text-align: center;
            background-color: blue;
            color: white;
            border-radius: 5px;
            cursor: pointer;
            width: 25%;
          }
        }
      }

      .colors {
        display: flex;
        flex-wrap: wrap;
      }
    }
  }
}
