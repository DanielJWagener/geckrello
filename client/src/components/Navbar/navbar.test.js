import { shallow } from "enzyme";
import React from "react";
import Navbar from "./navbar.component";
import { Navbar as NavbarPure } from "./navbar.component";
import Root from "../../Root";

it("renders Navbar component", () => {
  expect(shallow(<NavbarPure />).debug()).toMatchSnapshot();
});
