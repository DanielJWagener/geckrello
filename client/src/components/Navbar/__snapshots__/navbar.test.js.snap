// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar component 1`] = `
"<div className=\\"navbar\\">
   
  <Link to=\\"/\\" className=\\"navbar__link\\">
    <h1 className=\\"navbar__logo\\">
      Geckrello
    </h1>
  </Link>
  <a href=\\"/api/v1/users/logout\\" className=\\"navbar__button\\">
    Log Out
  </a>
</div>"
`;
